misc-scripts/xref_mapping/XrefParser/MGI_Desc_Parser.pm

```diff
@@ -18,135 +16,125 @@ limitations under the License.
 
 =cut
 
+=head1 NAME
+
+XrefParser::MGI_Desc_Parser
+
+=head1 DESCRIPTION
+
+A parser class to parse the MGI (descriptions) source. Creates 'MISC' xref using MGI accession with description and
+also creates the synonyms extracted from the pipe seperated synonym_field
+
+-species = mus_musculus
+-species_id = 10090
+-data_uri = http://www.informatics.jax.org/downloads/reports/MRK_List2.rpt
+-file_format = TSV
+-columns = [accession chromosome position start end strand label status marker marker_type feature_type synonym_field]
+
+=head1 SYNOPSIS
+
+  my $parser = XrefParser::MGI_Desc_Parser->new($db->dbh);
+  $parser->run({
+    source_id  => 58,
+    species_id => 10090,
+    files      => ["MRK_List2.rpt"],
+  });
+
+=cut
+
 package XrefParser::MGI_Desc_Parser;
 
 use strict;
 use warnings;
 use Carp;
-use File::Basename;
+use Text::CSV;
 
-use base qw( XrefParser::BaseParser );
+use parent qw( XrefParser::BaseParser );
+
+my $EXPECTED_NUMBER_OF_COLUMNS = 12;
+
+
+
+=head2
+
+The run method does the actual parsing and creation of xrefs and synonyms.
+Parser gets initialized as noted above and run is called from
+Bio::EnsEMBL::Production::Pipeline::Xrefs::ParseSource
+
+=cut
 
 sub run {
 
   my ($self, $ref_arg) = @_;
-  my $source_id    = $ref_arg->{source_id};
-  my $species_id   = $ref_arg->{species_id};
-  my $files        = $ref_arg->{files};
-  my $verbose      = $ref_arg->{verbose};
-  my $dbi          = $ref_arg->{dbi};
-  $dbi = $self->dbi unless defined $dbi;
-
-  if((!defined $source_id) or (!defined $species_id) or (!defined $files) ){
-    croak "Need to pass source_id, species_id and files as pairs";
+  my $source_id  = $ref_arg->{source_id};
+  my $species_id = $ref_arg->{species_id};
+  my $files      = $ref_arg->{files};
+  my $verbose    = $ref_arg->{verbose} // 0;
+  my $dbi        = $ref_arg->{dbi} // $self->dbi;
+
+  if ( ( !defined $source_id ) or
```

Comment:
Personally I would take the NOT( AND AND AND) version of this sort of logic, as it easier to process, but whatever